pkg/controllermanager/controller/project/project_activity_control_test.go

```diff
@@ -0,0 +1,196 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package project
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/gardener/gardener/pkg/apis/core"
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	"github.com/gardener/gardener/pkg/logger"
+	mockclient "github.com/gardener/gardener/pkg/mock/controller-runtime/client"
+	"github.com/golang/mock/gomock"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var (
+	namespaceName = "namespace"
+	projectName   = "name"
+	shootName     = "shoot"
+)
+
+var _ = Describe("Project Activity Reconcile", func() {
+	var (
+		project             *gardencorev1beta1.Project
+		shoot               *gardencorev1beta1.Shoot
+		shootWithoutProject *gardencorev1beta1.Shoot
+		errorShoot          *gardencorev1beta1.Shoot
+		namespace           *corev1.Namespace
+
+		partialObjectMetadata *metav1.PartialObjectMetadata
+
+		reconciler reconcile.Reconciler
+
+		request      reconcile.Request
+		errorRequest reconcile.Request
+
+		ctrl                   *gomock.Controller
+		k8sGardenRuntimeClient *mockclient.MockClient
+		ctx                    = context.TODO()
+
+		partialShootMetaList = &metav1.PartialObjectMetadataList{TypeMeta: metav1.TypeMeta{APIVersion: "core.gardener.cloud/v1beta1", Kind: "ShootList"}}
+	)
+
+	BeforeEach(func() {
+
+		project = &gardencorev1beta1.Project{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      projectName,
+				Namespace: namespaceName,
+				UID:       "1",
+			},
+			Status: gardencorev1beta1.ProjectStatus{
+				LastActivityTimestamp: &metav1.Time{Time: time.Date(1, 1, 1, 1, 1, 1, 1, time.UTC)},
+			},
+			Spec: gardencorev1beta1.ProjectSpec{
+				Namespace: &namespaceName,
+			},
+		}
+		shoot = &gardencorev1beta1.Shoot{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      shootName,
+				Namespace: namespaceName,
+				UID:       "1",
+			},
+		}
+		shootWithoutProject = &gardencorev1beta1.Shoot{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      shootName,
+				Namespace: "fake",
+				UID:       "1",
+			},
+		}
+		errorShoot = &gardencorev1beta1.Shoot{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      shootName,
+				Namespace: "error",
+				UID:       "1",
+			},
+		}
+		namespace = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{Name: namespaceName},
+		}
+		partialObjectMetadata = &metav1.PartialObjectMetadata{
+			ObjectMeta: metav1.ObjectMeta{Namespace: namespaceName},
+		}
+
+		logger.Logger = logger.NewLogger("info")
+		ctrl = gomock.NewController(GinkgoT())
+		k8sGardenRuntimeClient = mockclient.NewMockClient(ctrl)
```

Comment:
Can you double check the `EXPECT()` calls for this mock. I think there are a few leftovers after you removed the dependency to the stale project reconciler.

Reply:
Yep, there were some `EXPECT()s` that were not actually expected